tests/integration/conftest.py

```diff
@@ -125,14 +125,27 @@ def gateway_client_pod(client, worker_node, subnet_resource):
         client.delete(type(obj), obj.metadata.name, namespace=obj.metadata.namespace)
 
 
-async def wait_pod_ip(client, pod):
-    for _, obj in client.watch(
-        Pod,
-        namespace=pod.metadata.namespace,
-        fields={"metadata.name": pod.metadata.name},
-    ):
-        if obj.status.podIP:
-            return obj
+async def wait_pod_ips(client, pods):
```

Comment:
how about some docstring here to help indicate what this method does

```suggestion
async def wait_pod_ips(client, pods):
    """Returns a list of pods which have an ip address assigned."""

```

Reply:
You're dead right! I'll add both docstrings to the methods.